src/Visualization/Python/PlotPowerMonitors.py

```diff
@@ -0,0 +1,312 @@
+                and block_name in domain.block_groups[name]):
+            return i
+    return None
+
+
+def plot_power_monitors(volfiles: Union[spectre_h5.H5Vol,
+                                        Iterable[spectre_h5.H5Vol]],
+                        obs_id: int, tensor_components: Sequence[str],
+                        block_or_group_names: Sequence[str],
+                        domain: Union[Domain[1], Domain[2], Domain[3]]):
+    # One subplot per block or group
+    num_plots = len(block_or_group_names)
+    fig, axes = plt.subplots(nrows=1,
+                             ncols=num_plots,
+                             figsize=(num_plots * 4, 4),
+                             sharey=True,
+                             sharex=True,
+                             squeeze=False)
+
+    # Evaluate property cycles (by default this is just 'color'). We do multiple
+    # plotting commands (at least one per element), so we don't want matplotlib
+    # to cycle through the properties at every plotting command.
+    prop_cycle = {
+        key: cycle(values)
+        for key, values in plt.rcParams['axes.prop_cycle'].by_key().items()
+    }
+    props_dim = {
+        d: {key: next(values)
+            for key, values in prop_cycle.items()}
+        for d in range(domain.dim)
+    }
+
+    # Collect some reduction data for each subplot
+    num_elements = np.zeros(num_plots, dtype=int)
+    max_error = np.zeros((num_plots, domain.dim))
+
+    for element, tensor_data in iter_elements(volfiles, obs_id,
+                                              tensor_components):
+        # Find the subplot for this element's block, or skip the element if its
+        # block wasn't selected
+        subplot_index = find_block_or_group(element.id.block_id,
+                                            block_or_group_names, domain)
+        if subplot_index is None:
+            continue
+        num_elements[subplot_index] += 1
+        ax = axes[0][subplot_index]
+        # Compute power monitors and take L2 norm over tensor components
+        all_modes = [
+            np.zeros(element.mesh.extents(d)) for d in range(element.dim)
+        ]
+        for component in tensor_data:
+            modes = power_monitors(DataVector(component), element.mesh)
+            for d, modes_dim in enumerate(modes):
+                all_modes[d] += modes_dim**2
+        for d in range(element.dim):
+            all_modes[d] = np.sqrt(all_modes[d])
+            # Collect reduction data
+            # - We estimate the truncation error by just taking the highest
+            #   mode. This won't work well with filtering and should be improved
+            #   on the C++ side.
+            max_error[subplot_index][d] = max(max_error[subplot_index][d],
+                                              all_modes[d][-1])
+        # Plot
+        for d, modes_dim in enumerate(all_modes):
+            ax.semilogy(modes_dim, **props_dim[d], zorder=30 + d)
```

Comment:
Why 30? Just a large number?

Reply:
yea for the zorder only relative numbers matter